features/step_definitions/controllers-steps.js

```diff
@@ -207,18 +207,23 @@ Then('I send a HTTP {string} request with:', async function (method, dataTable)
     method,
     body: {
       jwt: this.sdk.jwt,
-      ...body,
+      ...body
     },
     headers: body.headers,
   };
 
-  delete body.headers;
+  body.headers = undefined;
 
-  const response = await requestPromise(options);
+  try {
+    const response = await requestPromise(options);
 
-  this.props.result = response.body.result;
-  this.props.response = response.body;
-  this.props.rawResponse = response;
+    this.props.result = response.body.result;
+    this.props.response = response.body;
+    this.props.rawResponse = response;
+  }
+  catch (err) {
+    this.props.error = err.error.error;
```

Comment:
If you are silently catch the error then you have to modify the step so it can accept a "successfully" modifier.  
In this way, you can ensure the request is successful with ` I successfully send a HTTP "post" request with:`.  
Previous usage of the step were assuming success so they should be updated accordingly.

Example here https://github.com/kuzzleio/kuzzle/blob/2-dev/features/step_definitions/controllers-steps.js#L12

Reply:
I fixed it, but instead of using `I send a (successfully ) ?HTTP {string} request with:` I replace the `successfully` by `bad`, that way I only need to change only my new test